l10n_th_sequence/models/ir_sequence.py

```diff
@@ -0,0 +1,133 @@
+from datetime import datetime
+
+import pytz
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class IrSequence(models.Model):
+    """
+    This sub-class is to refactor the ir.sequence by changing the inextensible
+    inner function `_interpolation_dict()` to a normal private class function.
+    This sub-class also adds support for `range_end_` prefix.
+    """
+
+    _inherit = "ir.sequence"
+
+    def _interpolation_dict(self):
+        now = range_end_date = range_date = effective_date = datetime.now(
+            pytz.timezone(self._context.get("tz") or "UTC")
+        )
+        if self._context.get("ir_sequence_date"):
+            effective_date = fields.Datetime.from_string(
+                self._context.get("ir_sequence_date")
+            )
+        if self._context.get("ir_sequence_date_range"):
+            range_date = fields.Datetime.from_string(
+                self._context.get("ir_sequence_date_range")
+            )
+        if self._context.get("ir_sequence_date_range_end"):
+            range_end_date = fields.Datetime.from_string(
+                self._context.get("ir_sequence_date_range_end")
+            )
+
+        sequences = {
+            "year": "%Y",
+            "month": "%m",
+            "day": "%d",
+            "y": "%y",
+            "doy": "%j",
+            "woy": "%W",
+            "weekday": "%w",
+            "h24": "%H",
+            "h12": "%I",
+            "min": "%M",
+            "sec": "%S",
+        }
+        res = {}
+        for key, fmt in sequences.items():
+            res[key] = effective_date.strftime(fmt)
+            res["range_" + key] = range_date.strftime(fmt)
+            res["range_end_" + key] = range_end_date.strftime(fmt)
+            res["current_" + key] = now.strftime(fmt)
+
+        return res
+
+    def _get_prefix_suffix(self):
```

Comment:
Please don't override this method, we need to always call super(). You can do something like this,
https://github.com/OCA/server-ux/pull/361/files#diff-9e9a96594f4c35da625b41834dc4f4ed874f65560d36af78fe8fbb22e04317a4R14-R17